src/sentry/api/endpoints/release_thresholds/utils/get_new_issue_counts.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List
+
+from django.db.models import CharField, Count, Q, Value
+
+from sentry.models.group import Group
+
+if TYPE_CHECKING:
+    from sentry.api.endpoints.release_thresholds.types import EnrichedThreshold
+
+
+def get_new_issue_counts(
+    organization_id: int, thresholds: List[EnrichedThreshold]
+) -> dict[str, int]:
+    """
+    constructs a query for each threshold, filtering on project
+    """
+    queryset = None
+    for t in thresholds:
+        query = Q(
+            project__organization__id=organization_id,
+            project__id=t.project_id,
+        )
+        if t.environment:
+            # If we're filtering on environment, just use the first_release/first_seen from the groupenvironment so we don't need to cross reference tables
+            query &= Q(
+                groupenvironment__first_release__version=t.release,
+                groupenvironment__first_seen__range=(t.start, t.end),
+                groupenvironment__environment__name=t.environment.get("name"),
+            )
+        else:
+            query &= Q(
+                first_release__version=t.release,
+                first_seen__range=(t.start, t.end),
+                groupenvironment__isnull=True,
+            )
+        qs = (
+            Group.objects.filter(query)
+            .values("project__id", "first_release__version", "groupenvironment")
```

Comment:
`project_id` instead of `project__id` to skip an unnecessary join

Reply:
I also don't think we need these values, since we've already done all our filtering, and we don't reference these columns for our use case. Could just leave this off right?

Reply:
fair
initially was thinking we could avoid pulling all the columns and at least pull just the potentially useful ones.

Since we're aggregating these by release version, i'll just keep that as our pulled value. better than pulling all the columns still i think 🤷